src/main/java/net/test/interceptor/TraceRetValueInterceptor.java

```diff
@@ -18,22 +19,31 @@
 
 public class TraceRetValueInterceptor {
 
+  private static String LOG_THRESHOLD_MILLISECONDS = "log_threshold_ms";
+
   private static List<String> KNOWN_ARGS = 
-    Arrays.asList(CommonActionArgs.IS_DATE_LOGGED);
+    Arrays.asList(CommonActionArgs.IS_DATE_LOGGED, LOG_THRESHOLD_MILLISECONDS);
 
   private CommonActionArgs commonActionArgs;
 
+  private final long logThresholdMs;
+
   public TraceRetValueInterceptor(String actionArgs, DefaultArguments defaults) {
-    Map<String, String> parsed = ArgUtils.parseOptionalArgs(KNOWN_ARGS, actionArgs);
+    ArgumentsCollection parsed = ArgUtils.parseOptionalArgs(KNOWN_ARGS, actionArgs);
     this.commonActionArgs = new CommonActionArgs(parsed, defaults);
+    this.logThresholdMs = parsed.parseLong(LOG_THRESHOLD_MILLISECONDS, 0);
   }
 
 
   @RuntimeType
   public Object intercept(@Origin Method method, @AllArguments Object[] allArguments, @SuperCall Callable<?> callable) throws Exception  {
+    long start = System.currentTimeMillis();
```

Comment:
Use the condition `(this.logThresholdMs == 0) ? 0 : System.currentTimeMillis();`

Reply:
Done!